examples/robot/lifelong_learning_bench/testalgorithms/rfnet/basemodel-sam.py

```diff
@@ -0,0 +1,209 @@
+        self.trainer = None
+
+        self.val_args = ValArgs(**kwargs)
+        label_save_dir = Context.get_parameters("INFERENCE_RESULT_DIR", "./inference_results")
+        self.val_args.color_label_save_path = os.path.join(label_save_dir, "color")
+        self.val_args.merge_label_save_path = os.path.join(label_save_dir, "merge")
+        self.val_args.label_save_path = os.path.join(label_save_dir, "label")
+        self.validator = Validator(self.val_args)
+
+    def get_weights(self):
+        return self.trainer.get_weight()
+    
+    def set_weights(self, weights):
+        self.trainer.set_weight(weights)
+        
+        epoch_num = 0
+        print("Total epoch: ", epoch_num)
+        loss_all = []
+        for epoch in range(epoch_num):
+            train_loss = self.trainer.my_training(epoch)
+            loss_all.append(train_loss)
+        with open('/home/shijing.hu/ianvs/project/ianvs/train_loss_2.txt', 'a+') as file:
+            np.savetxt(file, loss_all)
+        file.close
+
+    def train(self, train_data, valid_data=None, **kwargs):
+        self.trainer = Trainer(self.train_args, train_data=train_data)
+        print("Total epoches:", self.trainer.args.epochs)
+        loss_all = []
+        for epoch in range(
+                self.trainer.args.start_epoch,
+                self.trainer.args.epochs):
+            if epoch == 0 and self.trainer.val_loader:
+                self.trainer.validation(epoch)
+            loss = self.trainer.training(epoch)
+            loss_all.append(loss)
+            if self.trainer.args.no_val and (
+                    epoch %
+                    self.trainer.args.eval_interval == (
+                            self.trainer.args.eval_interval -
+                            1) or epoch == self.trainer.args.epochs -
+                    1):
+                # save checkpoint when it meets eval_interval or the training
+                # finished
+                is_best = False
+                self.train_model_url = self.trainer.saver.save_checkpoint({
+                    'epoch': epoch + 1,
+                    'state_dict': self.trainer.model.state_dict(),
+                    'optimizer': self.trainer.optimizer.state_dict(),
+                    'best_pred': self.trainer.best_pred,
+                }, is_best)
+
+        self.trainer.writer.close()
+        return self.train_model_url
+
+    def predict(self, data, **kwargs):
+        """
+        Use the RFNet model to predict at the edge 
+        """
+        if len(data) > 10:
+            print("predict start for big data")
+            my_kwargs = {'num_workers': self.val_args.workers, 'pin_memory': True}
+            _, _, self.validator.test_loader, _ = make_data_loader(self.val_args, test_data=data, **my_kwargs)
+        else:
+            print("predict start for small data")
+            if not isinstance(data[0][0], dict):
+                data = self._preprocess(data)
+                #print("predict starting 69")
```

Comment:
The lines for local debug should not be committed, these TEMP PRINT lines should be deleted.

Reply:
TEMP PRINT lines are removed.